MobileWallet/Common/NotificationManager.swift

```diff
@@ -95,4 +155,220 @@ class NotificationManager {
             }
         }
     }
+
+    func registerDeviceToken(_ deviceToken: Data) {
+        //Avoid duplicate requests
+        guard NotificationManager.isRegisterRequestInProgress == false else {
+            TariLogger.warn("Server request already in progress")
+            return
+        }
+
+        let apnsDeviceToken = deviceToken.map {String(format: "%02.2hhx", $0)}.joined()
+
+        TariLogger.verbose("Registering device token with public key")
+
+        guard let wallet = TariLib.shared.tariWallet else {
+            return TariLogger.error("Failed to get wallet", error: WalletErrors.walletNotInitialized)
+        }
+
+        let (pubKey, pubKeyError) = wallet.publicKey
+        guard pubKeyError == nil else {
+            return TariLogger.error("Failed to get wallet pub key", error: pubKeyError)
+        }
+
+        let (pubKeyHex, hexError) = pubKey!.hex
+        guard hexError == nil else {
+            return TariLogger.error("Failed to get wallet hex pub key", error: hexError)
+        }
```

Comment:
I saw these `guard ... == nil else {}` checks around the code, it's great safety measure but might be slightly redundant if we're sure that the existence of `TariLib.shared.tariWallet` implies the existence of the `pubKey` for instance.

Reply:
Agreed, gets a bit repetitive. Although pub key still comes from the FFI lib and I don't know 100% what goes on in that. But I'm happy to start force unwrapping these things if you think that's alright?

Reply:
Ok, let's leave it as it is & think about it and I'll check with David & Philip too.